### PR TITLE
[WIP] docs: add example for non disposable daemons

### PR DESCRIPTION
**Pull request by daviddias**
The non-disposable daemons option is currently not working. It errors silently.

Steps to reproduce:

- 1. clone this branch, npm install, go to examples/local-non-disposable and run npm install for it to link the main module
- 2. run ipfs init and jsipfs-init
- 3. node index.js

Result

image

It should spawn a go and js daemons on top of .ipfs and .jsipfs respectively and run the .id call without a problem.

### CODE REVIEW

**Comment by hugomrdias** [comment]
You need to init and start, check the link below
https://github.com/ipfs/js-ipfsd-ctl#disposable-vs-non-disposable-nodes

**Comment by hugomrdias** [comment]
I would think whoever introduce this behaviour, tried to make disposable fully automated and non-disposable the flexible version. You just need to pass `.spawn {init: true, start:true}`

It's just the way it works and is documented right now, we can make an issue about it and change the behaviour.

**Comment by diasdavid** [comment]
`You just need to pass .spawn {init: true, start:true}` This is not working for me either (even if I delete init, given that I already have my repo init'ed)

**Comment by diasdavid** [comment]
Can we treat this PR as the issue? Or you prefer me to open another issue?

File: src/factory-daemon.js [hunk 1]
```diff
@@ -126,9 +124,14 @@ class FactoryDaemon {
...
 
     series([
+
+      // TODO if init fails, check if it was because the
+      // repo was already was inited, if yes, continue
       (cb) => options.init
         ? node.init(options.initOptions, cb)
         : cb(null, node),
+      // TODO if start fails, check if it was because there
+      // is a daemon already running and connect to it instead
```
Comment (diasdavid) [inline src/factory-daemon.js]: Just needs a bunch of checks here :)

File: src/factory-daemon.js [hunk 2]
```diff
@@ -98,8 +98,6 @@ class FactoryDaemon {
 
     if (!options.disposable) {
       const nonDisposableConfig = clone(defaultConfig)
-      options.init = false
-      options.start = false
```
Comment (diasdavid) [inline src/factory-daemon.js]: this was what was making it not start even if passed the options

File: src/factory-daemon.js [hunk 3]
```diff
@@ -127,10 +125,28 @@ class FactoryDaemon {
...
+          }
+        })
         : cb(null, node),
       (cb) => options.start
-        ? node.start(options.args, cb)
+        ? node.start(options.args, (err) => {
+          if (err) {
+            console.log(err)
+            // TODO if start fails, check if it was because there
+            // is a daemon already running and connect to it instead
```
Comment (diasdavid) [inline src/factory-daemon.js]: Would appreciate some brainstorming on this one. Would the best just to move this func out https://github.com/ipfs/js-ipfsd-ctl/blob/master/src/ipfsd-daemon.js#L229-L234 and if the daemon is running, catch the addr from the `api` file in the repo?

@hugomrdias?

**Comment by hugomrdias** [comment]
@diasdavid can you check https://github.com/ipfs/js-ipfsd-ctl/pull/308 related to this.

**Comment by hugomrdias** [comment]
doesn't apply anymore version 1.0.0 fixed this problem.

**Pull request closed without merging**